### PR TITLE
[CONTINT-3679] Add SBOM check tests and propagate the config component to the sbom scanner 

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR adds unit tests to the SBOM check. Because many different parts of the sbom scanner use `pkg/config`, I also replaced part of it by `config.Component`

### Describe how to test/QA your changes
TestSBOM e2e test should pass.

### CODE REVIEW

**Review by vickenty — APPROVED** [review]

**Review by GustavoCaso — APPROVED** [review]
This is fantastic @AliDatadog 🎉 

Thanks you so much for taking the time to write the tests and propagate the config component 😄

File: pkg/sbom/collectors/containerd/containerd.go [hunk 1]
```diff
@@ -40,7 +41,7 @@ func (r ScanRequest) Collector() string {
 
 // Type returns the scan request type
 func (r ScanRequest) Type() string {
-	if config.Datadog.GetBool("sbom.container_image.use_mount") {
+	if pkgconfig.Datadog.GetBool("sbom.container_image.use_mount") {
```
Comment (adel121) [inline pkg/sbom/collectors/containerd/containerd.go:49]: Do you think we can eliminate this call here? 

This is the only line in this file where `pkgconfig` is used. 

How about the following:
- Storing the config component as a dependency in `Scanner`, and passing the config down to `Type` method of `ScanRequest`
- Adding a `UseMount` property to the scanner struct, which is initialised to `sbom.container_image.use_mount` using the config component, and then we can pass that property directly to `Type`

My intention is to fully eliminate the usage of `pkg/config` here.

WDYT?

Comment (AliDatadog) [reply pkg/sbom/collectors/containerd/containerd.go:49]: I think it should work, nice catch 👍

**Review by adel121 — APPROVED** [review]

**Pull request merged**